### PR TITLE
split classes into separate files

### PR DESCRIPTION
**Pull request by rohitsangwan01**
This PR separates all classes into individual files, making it easier to support additional modules. It also simplifies the review process for specific modules and enhances readability for individual files. While these benefits may be subjective

### CODE REVIEW

File: lib/src/bluez_client.dart [hunk 1]
```diff
@@ -1941,11 +238,11 @@ class BlueZClient {
     return BlueZDevice(this, object);
   }
 
-  bool _isDevice(_BlueZObject object) {
+  bool _isDevice(BlueZObject object) {
     return object.interfaces.containsKey('org.bluez.Device1');
   }
 
-  List<BlueZGattService> _getGattServices(DBusObjectPath parentPath) {
```
Comment (robert-ancell) [inline lib/src/bluez_client.dart:1948]: This was previously private API that has now been exposed. Is there a suitable technique for keeping this private?

**Review by robert-ancell — CHANGES_REQUESTED** [review]
Thanks for working on this. One reason this hasn't been done before is there is some private API that we don't want to expose to users - this change has exposed it. Are there other Dart techniques we can use to keep it private? If part of this PR can land without exposing the API please propose that and we can land that part now.

**Comment by rohitsangwan01** [comment]
@robert-ancell i have used `part/partOf` for splitting this single files into multiple files, and still have private apis across all files,

There is one more option, For example we can split `BluezClient` into two classes, like this

Within same file `bluez_client.dart`

```dart
class BlueZClient {
  // main class with all the implementation..
  final _objects = <DBusObjectPath, BlueZObject>{};
}

// This wont be exposed outside of plugin
extension BluezClientPrivateExtension on BlueZClient {
   // This class have all the apis we want to access from other classes like `getGattServices`
  // These api will be public for the plugin

  List<BlueZGattService> getGattServices(DBusObjectPath parentPath) {
    var services = <BlueZGattService>[];
    for (var object in _objects.values) {
      if (object.path.isInNamespace(parentPath) && _isGattService(object)) {
        services.add(BlueZGattService(this, object));
      }
    }
    return services;
  }
}
```

And in `bluez.dart` file, we can expose only `BluezClient` class

```dart
export 'src/bluez_client.dart' show BlueZClient;
```

So this way, we can still keep the BluezClient apis private, and also exposed the required apis as public within plugin only


Screenshot 2024-09-23 at 9 45 20 AM

**Comment by robert-ancell** [comment]
Reading up on `part` it seems this is discouraged (https://dart.dev/guides/libraries/create-packages). So it looks like extensions at a good way to resolve this!

**Comment by rohitsangwan01** [comment]
@robert-ancell All done

**Review by robert-ancell — APPROVED** [review]
Thanks for this!

**Pull request merged**